extends Node2D

@export var mTileSize: int = 16
@export var mMapWidth: int = 200
@export var mMapHeight: int = 200

@onready var mTileMapLayer: TileMapLayer = get_node("../TileMapLayer")  # ğŸ”¹ ë¶€ëª¨ ë…¸ë“œì—ì„œ ìë™ìœ¼ë¡œ `TileMap`ì„ ì°¾ìŒ
@onready var mPlayer: CharacterBody2D = get_node("../Player")

var mMapMinX
var mMapMaxX
var mMapMinY
var mMapMaxY

func _ready():
	# ì•± ì „ì²´ì˜ ê¸°ë³¸ í´ë¦¬ì–´ ì»¬ëŸ¬ë¥¼ ê²€ì€ìƒ‰ìœ¼ë¡œ ì„¤ì •
	RenderingServer.set_default_clear_color(Color.BLACK)
	CalculateMapBoundaries()
	SetPlayerStartPosition()

func CalculateMapBoundaries():
	mMapMinX = position.x
	mMapMaxX = position.x + (mMapWidth * mTileSize) - mTileSize
	mMapMinY = position.y
	mMapMaxY = position.y + (mMapHeight * mTileSize) - mTileSize

# ğŸ”¹ í”Œë ˆì´ì–´ ìœ„ì¹˜ë¥¼ ë§µ ì¤‘ì•™ì— ë°°ì¹˜
func SetPlayerStartPosition():
	var startX = float(mMapWidth * mTileSize) / 2  # ì¤‘ì•™ X ìœ„ì¹˜
	var startY = float(mMapHeight * mTileSize) / 2  # ì¤‘ì•™ Y ìœ„ì¹˜
	#mPlayer.position = Vector2(startX, startY)
	mPlayer.position = Vector2.ZERO
	print("ğŸ”¹ í”Œë ˆì´ì–´ ì‹œì‘ ìœ„ì¹˜:", mPlayer.position)
